### PR TITLE
Rewrote ThermalHouseTest from groovy to scala

### PR DESCRIPTION
**Pull request by pierrepetersmeier**
ThermalHouseTest convert from groovy to scala.

This PR is related to Issue https://github.com/ie3-institute/simona/issues/646

Edit: Please merge #873 first!

### CODE REVIEW

File: src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        "Thermal House",
+        null,
+        getQuantity(1.0, StandardUnits.THERMAL_TRANSMISSION),
+        getQuantity(2741.5, StandardUnits.HEAT_CAPACITY),
+        getQuantity(0, CELSIUS), // stub
+        getQuantity(18, CELSIUS),
+        getQuantity(22, CELSIUS),
+      )
+
+      val thermalHouse = buildThermalHouse(18, 22)
```
Comment (danielfeismann) [inline src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:128]: This can as well shortend by using the method from HpInputTestData

**Review by danielfeismann — CHANGES_REQUESTED** [review]
Thanks for converting this. Just some little points.

File: src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import squants.time._
+import squants.{Energy, Temperature}
+import tech.units.indriya.quantity.Quantities.getQuantity
+import tech.units.indriya.unit.Units.CELSIUS
+
+import java.util.UUID
+
+class ThermalHouseSpec extends UnitSpec with ThermalHouseTestData {
+
+  implicit val tolerance: Temperature = Celsius(0.0001)
```
[the remark this answers is not in the document]
Comment (pierrepetersmeier) [reply src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:23]: Thanks :)

File: src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      "Thermal House",
+      null,
+      getQuantity(1.0, StandardUnits.THERMAL_TRANSMISSION),
+      getQuantity(10.0, StandardUnits.HEAT_CAPACITY),
+      getQuantity(0, CELSIUS), // stub
+      getQuantity(upperBoundaryTemperature, CELSIUS),
+      getQuantity(lowerBoundaryTemperature, CELSIUS),
+    )
+    ThermalHouse(thermalHouseInput)
+  }
```
[the remark this answers is not in the document]
Comment (pierrepetersmeier) [reply src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:41]: Then you were right after all. Sorry :)

[the remark this answers is not in the document]
Comment (danielfeismann) [reply src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:41]: No worries :)

**Comment by pierrepetersmeier** [comment]
@danielfeismann Could you take a look at the test: 'Check build method'? Am I comparing the correct thing there? Because thermalHouse and thermalHouseInput do not match. Thanks :).

File: src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala [hunk 4]
```diff
@@ -100,26 +93,26 @@ class ThermalHouseSpec extends UnitSpec with HpInputTestData {
...
+      (thermalHouseTest.ethCapa * Temperature(
         1,
         Celsius,
-      )).toKilowattHours shouldBe thermalHouseInput.getEthCapa
-        .to(StandardUnits.HEAT_CAPACITY)
-        .getValue
-        .doubleValue
-      thermalHouse.lowerBoundaryTemperature shouldBe Temperature(18, Celsius)
-      thermalHouse.upperBoundaryTemperature shouldBe Temperature(22, Celsius)
+      )).toKilowattHours shouldBe 2741.5
```
Comment (danielfeismann) [inline src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:107]: Rather pedantic, I know, but I would prefer to have the unit here tested as well => Quantity.getQuanties(1d, PowerSystemUnits.WATTHOUR). Then you can use should approximate as well.
Same for the others.
If you do so, there will be a conflict of different types as there are `squants.Energy` (used in SIMONA) and `ComparableQuantity[javax.measure.quantity.Energy]` (used in PSDM). But if you write one of them not only as Energy but e.g. as squants.Energy, this should work.

File: src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala [hunk 5]
```diff
@@ -20,19 +19,13 @@ class ThermalHouseSpec extends UnitSpec with HpInputTestData {
...
-      upperBoundaryTemperature: Double,
-  ): ThermalHouse = {
-    thermalHouse(lowerBoundaryTemperature, upperBoundaryTemperature)
-  }
-
   "ThermalHouse" should {
     "Functions testing inner temperature work as expected" in {
-      val thermalHouse = buildThermalHouse(18, 22)
+
+      val thermalHouseTest = thermalHouse(18, 22)
```
Comment (danielfeismann) [inline src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:25]: Maybe thermalTestHouse? or just house?

**Review by danielfeismann — CHANGES_REQUESTED** [review]
Thanks! Just two minor points I would like to have changed. :)

File: src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+      thermalHouseTest.id shouldBe thermalHouseInput.getId
+      thermalHouseTest.operatorInput shouldBe thermalHouseInput.getOperator
+      thermalHouseTest.operationTime shouldBe thermalHouseInput.getOperationTime
+      thermalHouseTest.bus shouldBe null
+      thermalHouseTest.ethLosses.toWattsPerKelvin shouldBe 1000.0
+      (thermalHouseTest.ethCapa * Temperature(
+        1,
+        Celsius,
+      )).toKilowattHours shouldBe 2741.5
```
Comment (danielfeismann) [inline src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:107]: Be careful here, original test is with 1 Kelvin which is different to what happens if one uses 1 Celsius.

Comment (pierrepetersmeier) [reply src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:107]: Yeah Thanks. :)

**Review by danielfeismann — APPROVED** [review]
LGTM

**Pull request merged**